x/ibc/23-commitment/merkle.go

```diff
@@ -111,14 +111,8 @@ func ApplyPrefix(prefix PrefixI, path string) (Path, error) {
 		return Path{}, errors.New("prefix can't be empty")
 	}
 
-	// Split paths by the separator
-	pathSlice := strings.Split(path, "/")
-	keyPath := merkle.KeyPath{}
-	commitmentPath := NewPath(pathSlice)
-
-	keyPath = keyPath.AppendKey(prefix.Bytes(), merkle.KeyEncodingURL)
-	commitmentPath.KeyPath = append(keyPath, commitmentPath.KeyPath...)
-	return commitmentPath, nil
+	path = fmt.Sprintf("%s%s", string(prefix.Bytes()), path)
+	return NewPath([]string{types.ModuleName, path}), nil
```

Comment:
`prefix.Bytes()` is `ModuleName` now, but it's `SubModuleName` in this version.

Reply:
we can undo these changes and open a PR or push them directly to ICS23

Reply:
The same changes has be commited in `cwgoes-ibc-demo-fixes` branch. Need to push to ICS23 now?

Reply:
Already fixed in the upstream branch